### PR TITLE
Feat(FWS-42): Ability to filter in the admin panel

### PR DESCRIPTION
**Pull request by emilegirardGit**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-42

What is the ticket about and why are we making this change?

This ticket was to make the admin panel easier to get the information needed by filtering the result.

What are the various changes and what other modules do those changes affect?

-Added filter component for reusability
-Added filter to accounts
-Added filter to appointments

UI Changes:
account
appointments

### CODE REVIEW

File: fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js [hunk 1]
```diff
@@ -1,80 +1,87 @@
...
+      })
+      .then((data) => {
+          setAccounts(data);
+      })
+      .catch((error) => {
+          console.log(error);
+      });
+  };
+
+    function onInputChange(label, value) {
```
Comment (AlexeiTimbro) [inline fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js:70]: Could you elaborate specifically on what the onInputChange function does? and how it works.

**Review by AlexeiTimbro — APPROVED** [review]
Great work Emile and I added a question about a specific part of the code down below. Answer it before you merge. Everything looks good.

**Review by Abdur-Siddiqui — APPROVED** [review]
Looks Good, good job on adding an important  feature

File: fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js (hunk 1 above)
Comment (emilegirardGit) [reply fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js:70]: onInputChange takes whats in the filter and send the a request to the API with query params.

Comment (emilegirardGit) [reply fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js:70]: And it keep track of what is inside of the filter with the usesate so it doesn't get remove when they close it.

File: fitwsarah_frontend/src/views/AdminPanelPage/AdminAppointments.js [hunk 2]
```diff
@@ -1,51 +1,40 @@
...
+
+    const [searchTerm, setSearchTerm] = useState([["appointmentid",""], ["accountid",""], ["status",""]]);
+
+    const labels = ["Appointment ID", "Account ID", "Status"];
 
     useEffect(() => {
-        if (isAuthenticated) {
-            const getAccessToken = async () => {
-                try {
-                    const token = await getAccessTokenSilently({
```
Comment (Eli293) [inline fitwsarah_frontend/src/views/AdminPanelPage/AdminAppointments.js:31]: Is there a reason why we are not using getAccessTokenSilently here?

Comment (emilegirardGit) [reply fitwsarah_frontend/src/views/AdminPanelPage/AdminAppointments.js:31]: This is getting the access token form the AuthUtils file it just take less space in the code when you need the access token in your views.

**Pull request merged**